src/rebar_prv_common_test.erl

```diff
@@ -37,54 +38,352 @@ init(State) ->
+    io_lib:format("Bad value for dir or suite in opts: ~p", [Opts]);
 format_error({failures_running_tests, {Failed, AutoSkipped}}) ->
     io_lib:format("Failures occured running tests: ~b", [Failed+AutoSkipped]);
 format_error({error_running_tests, Reason}) ->
     io_lib:format("Error running tests: ~p", [Reason]);
-format_error({error_processing_options, Reason}) ->
-    io_lib:format("Error processing options: ~p", [Reason]).
+format_error({error, Reason}) ->
+    io_lib:format("Unknown error: ~p", [Reason]).
+
+%% ===================================================================
+%% Internal functions
+%% ===================================================================
+
+run_test(State, Opts) ->
+    {RawOpts, _} = rebar_state:command_parsed_args(State),
+    Result = case proplists:get_value(verbose, RawOpts, false) of
+        true  -> run_test(Opts);
+        false -> run_test_quiet(Opts)
+    end,
+    ok = rebar_prv_cover:maybe_write_coverdata(State, ?PROVIDER),
+    case Result of
+        ok    -> {ok, State};
+        Error -> Error
+    end.
+
+run_test(Opts) -> handle_results(ct:run_test(Opts)).
 
-run_test(CTOpts, true) ->
-    handle_results(ct:run_test(CTOpts));
-run_test(CTOpts, false) ->
+run_test_quiet(Opts) ->
     Pid = self(),
-    LogDir = proplists:get_value(logdir, CTOpts),
+    LogDir = proplists:get_value(logdir, Opts),
     erlang:spawn_monitor(fun() ->
         {ok, F} = file:open(filename:join([LogDir, "ct.latest.log"]),
                             [write]),
         true = group_leader(F, self()),
-        Pid ! ct:run_test(CTOpts)
+        Pid ! ct:run_test(Opts)
     end),
-    receive Result -> handle_quiet_results(CTOpts, Result) end.
+    receive Result -> handle_quiet_results(Opts, Result) end.
+
+handle_results(Results) when is_list(Results) ->
+    Result = lists:foldl(fun sum_results/2, {0, 0, {0,0}}, Results),
+    handle_results(Result);
+handle_results({_, Failed, {_, AutoSkipped}})
+  when Failed > 0 orelse AutoSkipped > 0 ->
+    ?PRV_ERROR({failures_running_tests, {Failed, AutoSkipped}});
+handle_results({error, Reason}) ->
+    ?PRV_ERROR({error_running_tests, Reason});
+handle_results(_) ->
+    ok.
+
+sum_results({Passed, Failed, {UserSkipped, AutoSkipped}},
+            {Passed2, Failed2, {UserSkipped2, AutoSkipped2}}) ->
+    {Passed+Passed2, Failed+Failed2,
+     {UserSkipped+UserSkipped2, AutoSkipped+AutoSkipped2}}.
+
+handle_quiet_results(_, {error, _} = Result) ->
+    handle_results(Result);
+handle_quiet_results(_, {'DOWN', _, _, _, Reason}) ->
+    handle_results(?PRV_ERROR(Reason));
+handle_quiet_results(CTOpts, Results) when is_list(Results) ->
+    _ = [format_result(Result) || Result <- Results],
+    case handle_results(Results) of
+        ?PRV_ERROR({failures_running_tests, _}) = Error ->
+            LogDir = proplists:get_value(logdir, CTOpts),
+            Index = filename:join([LogDir, "index.html"]),
+            ?CONSOLE("Results written to ~p.", [Index]),
```

Comment:
this is actually a non-problem. the ct output we suppress has the index url in it. we only print it when it wouldn't otherwise be printed